client/question/components/LearnerViewDirective.js

```diff
@@ -734,25 +798,41 @@ tie.directive('learnerView', [function() {
         };
 
         /**
+         * Displays congratulations when the question is complete.
+         * Also sends a QuestionCompleteEvent to the backend.
+         */
+        $scope.completeQuestion = function() {
+          congratulatoryFeedback.clear();
+          congratulatoryFeedback.appendTextParagraph(
+              "Good work! You've completed this question.");
+          congratulatoryFeedback.appendTextParagraph(
+              'Click the "Next" button to the right to proceed to the ' +
+              'next question.');
+          $scope.nextButtonIsShown = true;
+          $scope.feedbackStorage.push({
+            feedbackParagraphs: congratulatoryFeedback.getParagraphs()
+          });
+          EventHandlerService.createQuestionCompleteEvent(
+            SessionIdService.getSessionId(), $scope.currentQuestionId,
+            'QUESTION_VERSION');
+        };
+
+        /**
          * Sets the feedbackStorage property to the appropriate text according
          * to the feedback passed into the function.
          *
          * @param {Feedback} feedback
+         * @param {string} code
          */
-        var setFeedback = function(feedback) {
+        $scope.setFeedback = function(feedback, code) {
+          EventHandlerService.createCodeSubmitEvent(
+            SessionIdService.getSessionId(),
+            feedback.getParagraphsAsListOfDicts(), 'ERROR CATEGORY',
```

Comment:
For now, omit the 'ERROR CATEGORY' argument altogether. We'll add it back later if needed. (Same reason as above.)

Reply:
why not just create feedback.getErrorCategory() and return null, for consistency with the solution above?

I'd rather do that than rearchitect EventHandlerService twice.

Reply:
That's fine too, yes.